dynamic/discovery/discovery.go

```diff
@@ -89,6 +89,23 @@ func (rm *ResourceMap) GetKind(apiVersion, kind string) (result *APIResource) {
 	return gv.kinds[kind]
 }
 
+func (rm *ResourceMap) HasSubresource(parentResource *APIResource, subresourceKey string) bool {
+	rm.mutex.RLock()
```

Comment:
Another benefit of denormalizing the knowledge of whether /status exists into every APIResource object is that we won't have to take a lock to read it.